### PR TITLE
Preparations for move to new team

### PR DESCRIPTION
**Pull request by osfameron**
This includes
- script to create a mailmerge to advise people of change, and give an opt-out link
- an informational page about the transition...
- ...if called with opt-out link, then provide the option of a "delete my account" button
- minor tweaks to retirement banner and text

The deletion info is stored in a new table, and will require processing (either manual or automated TODO) to ensure that we only transfer the accounts that have not opted out.

### CODE REVIEW

File: templates/tcs.tt [hunk 1]
```diff
@@ -0,0 +1,52 @@
+[% IF action %]
+
+    Thanks!  We've registered your intention to [% action %] your account and hassles
```
Comment (dracos) [inline templates/tcs.tt]: I believe there is an HTML injection here.

File: templates/tcs.tt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        and to develop new features.</p>
+
+        <p>The plan is that the new team will run the site with all of the most important
+    principles that you signed up for:</p>
+    </div>
+
+    <ul class="tc-promises">
+        <li><img src="/globe.png" width="64" height="64" alt="Globe"><br>HassleMe will<br>always be free</li>
+        <li><img src="/user_circle.png" width="64" height="64" alt="User Circle"><br>HassleMe will never<br>sell your email address,<br>or send you spam</li>
+        <li><img src="/heart.png" width="64" height="64" alt="Heart"><br>HassleMe will<br>be run by people<br>who love it.</li>
```
Comment (dracos) [inline templates/tcs.tt]: Please drop the content of the `alt` attributes – things for decoration should have `alt=""` :)

File: templates/tcs.tt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    </div>
+
+    <div class="tc-decision">
+    [% IF email  %]
+        <p>If you’re happy with that, you don’t need to do anything. Your service will seamlessly continue, and your Hassles will be delivered as requested.</p>
+
+        <p>If, however, you’d prefer to not to be part of the new HassleMe, you can delete your HassleMe account and all associated Hassles:</p>
+
+        <p class="tc-decision__final">
+            <a class="tc-button" href="/TT/[% token %]/delete">Delete my HassleMe account</a>
```
Comment (dracos) [inline templates/tcs.tt]: Another one here.

File: templates/tcs.tt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        </p>
+    [% ELSE %]
+        <p>You should have received an email from us with a personalized link to be able to delete
```
Comment (dracos) [inline templates/tcs.tt]: "continue or delete your data" or similar? Don't want to imply deletion is all that can happen, I mean (even though I realise continuing is doing nothing.)

File: web/hassleme.cgi [hunk 5]
```diff
@@ -412,6 +423,34 @@ EOF
...
 }
+
+sub fn_terms {
+    my $q = shift;
+    hassle_header($q);
+    my $vars;
+    my $token = $q->param('token');
+    my $email;
+
+    if (my $checked = check_token($q->param('token'))) {
```
Comment (dracos) [inline web/hassleme.cgi]: You put token in `$token` a couple of lines above.

File: web/hassleme.cgi [hunk 6]
```diff
@@ -412,6 +423,34 @@ EOF
...
+        if ($checked =~/^TC.(.*)/) {
+            ($email) = dbh()->selectrow_array("select email from recipient where id = ?", {}, $1);
+            $vars->{token} = $token;
+            $vars->{email} = $email;
+        }
+    }
+
+    if (my $action = $q->param('action')) {
+        dbh()->do('insert into tc (email, action) values (?, ?)',
+                    {}, $email, $action);
```
Comment (dracos) [inline web/hassleme.cgi]: This doesn't check that `$email` exists.

File: templates/tcs.tt (hunk 3 above)
Comment (dracos) [inline templates/tcs.tt]: Ah, no, this one isn't, because token is checked.

**Comment by osfameron** [comment]
Thanks @dracos!

I'll review and correct later (probably tonight) and if all ok deploy tomorrow (pushing back release date to 10th March)

File: templates/tcs.tt (hunk 4 above)
Comment (osfameron) [inline templates/tcs.tt]: Hmm, though the backend does have the ability to register a "keep" message, that's been removed from the frontend, so the only way that the personalized link is different from the current page *is* the ability to delete your data.

**Comment by osfameron** [comment]
Thanks, addressed issues hopefully, will squash and deploy tomorrow (lunch/afternoon) unless any more comments!

**Pull request merged**